extensions/kubectl-cli/src/extension.spec.ts

```diff
@@ -0,0 +1,189 @@
+    })
+    .mockResolvedValueOnce({
+      stderr: '',
+      stdout: '/path1/to/kubectl\n/path2/to/kubectl',
+      command: 'kubectl version --client=true -o=json',
+    });
+
+  const deferred = new Promise<void>(resolve => {
+    vi.mocked(extensionApi.cli.createCliTool).mockImplementation(() => {
+      resolve();
+      return {} as extensionApi.CliTool;
+    });
+  });
+
+  KubectlExtension.activate();
+
+  return deferred.then(() => {
+    expect(extensionApi.cli.createCliTool).toBeCalled();
+    expect(extensionApi.cli.createCliTool).toBeCalledWith(
+      expect.objectContaining({
+        name: 'kubectl',
+        version: '1.28.3',
+        path: '/path1/to/kubectl',
+      }),
+    );
+  });
+});
+
+test('kubectl CLI tool not registered when not detected', async () => {
+  vi.mocked(extensionApi.process.exec).mockRejectedValue(new Error('Error running version command'));
+  const deferred = new Promise<void>(resolve => {
+    vi.spyOn(console, 'log').mockImplementation(() => {
+      resolve();
+    });
+  });
+
+  KubectlExtension.activate();
+
+  return deferred.then(() => {
+    expect(console.log).toBeCalled();
+    expect(console.log).toBeCalledWith(expect.stringContaining('Cannot detect kubectl CLI tool:'));
+  });
+});
+
+test('kubectl CLI tool not registered when version json stdout cannot be parsed', async () => {
+  vi.mocked(extensionApi.process.exec).mockResolvedValueOnce({
+    stderr: '',
+    stdout: `{${JSON.stringify(jsonStdout)}`,
+    command: 'kubectl version --client=true -o=json',
+  });
+
+  const deferred = new Promise<void>(resolve => {
+    vi.spyOn(console, 'log').mockImplementation((message: string) => {
+      log(message);
+      resolve();
+    });
+  });
+
+  KubectlExtension.activate();
+
+  return deferred.then(() => {
+    expect(console.log).toBeCalled();
+    expect(console.log).toBeCalledWith(
+      expect.stringContaining('Cannot detect kubectl CLI tool: SyntaxError: Unexpected token {'),
+    );
+  });
+});
+
+test('kubectl CLI tool not registered when version cannot be extracted from object', async () => {
+  const wrongJsonStdout = {
+    clientVersion: {
+      ...jsonStdout.clientVersion,
+    },
+  };
+  delete (wrongJsonStdout.clientVersion as any).gitVersion;
+  vi.mocked(extensionApi.process.exec).mockResolvedValueOnce({
+    stderr: '',
+    stdout: JSON.stringify(wrongJsonStdout),
+    command: 'kubectl version --client=true -o=json',
+  });
+
+  const deferred = new Promise<void>(resolve => {
+    vi.spyOn(console, 'log').mockImplementation((message: string) => {
+      log(message);
+      resolve();
+    });
+  });
+
+  KubectlExtension.activate();
+
+  return deferred.then(() => {
+    expect(console.log).toBeCalled();
+    expect(console.log).toBeCalledWith(
+      expect.stringContaining('Cannot detect kubectl CLI tool: Error: Cannot extract version from stdout'),
+    );
+  });
+});
+
+test('kubectl CLI tool not registered when path cannot be found', async () => {
```

Comment:
This and the above case are a bit odd to me, because we know kubectl _is_ installed but we're reporting it's not because we either can't tell what version it is or where it is. Shouldn't we be reporting partial data or giving the error with the data we do know? Or maybe the API doesn't support that yet?

Reply:
> @benoitf: 
>
> no, we only register for now if it has been found
> 
> not found : nothing
> found: provides the path of the CLI and provides the version

@deboer-tim it was a request from @benoitf (see resolved conversations above). Compose extension seems to register it no matter if it found or not.